providers/base/units/usb/usb-c.pxu

```diff
@@ -208,3 +208,129 @@ _steps:
+plugin: user-interact-verify
+user: root
+estimated_duration: 300.0
+command:
+   modprobe g_serial
+_steps:
+ 1. Press enter to probe USB OTG g_serial
+ 2. Connect a USB cable from an Ubuntu host to the USB OTG (it's the USB type C connector) port on DUT.
+ 3. On the host, to check if ttyACM0 has been probed
+    $ ls /dev/ttyACM0
+ 4. On the host, listen to the ttyACM0 port as a receive side.
+    $ sudo su
+    $ cat /dev/ttyACM0
+ 5. On DUT, send a string via /dev/ttyGS0 as a send side.
+    $ sudo su
+    $ echo 123 > /dev/ttyGS0
+ 6. Check if received the string "123" on the host side.
+ 7. Check the string sending from the host to DUT by repeating steps 4-6, but swap the send and receive sides.
+_verification:
+    Does string send and receive work?
+flags: preserve-locale also-after-suspend
+imports: from com.canonical.plainbox import manifest
+requires: manifest.has_usbc_otg == 'True'
+
+id: usb-c-otg/g_serial-cleanup
+_summary: Cleanup USB OTG serial interface setup after serial device test
+plugin: shell
+after: usb-c-otg/g_serial
+category_id: com.canonical.plainbox::usb
+command:
+  echo "Removing g_serial module..."
+  modprobe -r g_serial
+user: root
+estimated_duration: 1s
+flags: also-after-suspend
+
+id: usb-c-otg/g_mass_storage
+_summary: Check DUT can be detected as a mass storage device
+_purpose:
+  Check that after connecting the device under test (DUT) to another device
+  (host), DUT can be detected as a mass storage device by the host.
+_steps:
+  1. Press Enter to setup a 16 MB FAT32 image on the device.
+  2. From an Ubuntu host, connect a cable to the USB OTG port 
+     (it's the USB type C connector) of the DUT.
+_verification:
+  The host detects and mounts a mass storage device. It has read and write 
+  permissions on it.
+plugin: user-interact-verify
+command:
+  echo "Creating 16 MB image file..."
+  dd if=/dev/zero of="${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img bs=1M count=16
+  echo "Formatting image file as FAT32..."
+  mkdosfs -F 32 "${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img
+  echo "loading g_mass_storage module..."
+  modprobe g_mass_storage file="${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img stall=0
+user: root
+category_id: com.canonical.plainbox::usb
+estimated_duration: 5m
+flags: preserve-locale also-after-suspend
+imports: from com.canonical.plainbox import manifest
+requires: manifest.has_usbc_otg == 'True'
+
+id: usb-c-otg/g_mass_storage-cleanup
+_summary: Cleanup mass storage setup after mass storage device test
+plugin: shell
+after: usb-c-otg/g_mass_storage
+command:
+  echo "Removing g_mass_storage module..."
+  modprobe -r g_mass_storage
+  if [ -f "${PLAINBOX_SESSION_SHARE}/checkbox-mass-storage-test.img" ]; then
+    echo "Removing image file..."
+    rm "${PLAINBOX_SESSION_SHARE}"/checkbox-mass-storage-test.img
+  fi
+user: root
+estimated_duration: 1s
+category_id: com.canonical.plainbox::usb
+flags: also-after-suspend
```

Comment:
Similar to my comment above for the serial test, you will have to set things manually with the `siblings` field if you want to point to the right `after` job.